Project/Source/lcdfunctions.c

```diff
@@ -0,0 +1,264 @@
+	
+	bullet.positionX = x;
+	bullet.positionY = y;
+	bullet.direction = player.direction;
+	bullet.life = 5;
+	
+	return bullet;
+}
+
+void BulletMove(BulletPosition *bullet){			/* pomicanje metka */			/* prima  poziciju metka */
+	int x = bullet->positionX,y = bullet->positionY;
+	
+	BulletRemove(*bullet);
+	LCD_SetTextColor(LCD_COLOR_BLACK);
+	
+	if(bullet->direction==1){							/* smjer 1=dolje, 2=lijevo, 3=gore, 4=desno */
+		y+=yb;																																					
+	}
+	else if(bullet->direction==2 && x-xb > 0){
+		x-=xb;
+	}
+	else if(bullet->direction==3 && y-yb > 0){
+		y-=yb;
+	}
+	else{
+		x+=xb;										
+	}
+	
+	LCD_DrawFullCircle(x, y, rb);
+	
+	bullet->positionX = x;
+	bullet->positionY = y;
+}
+
+void BulletRemove(BulletPosition bullet){					/* brise metak */
+	LCD_SetTextColor(LCD_COLOR_WHITE);		/* prima  poziciju metka */
+	LCD_DrawFullCircle(bullet.positionX, bullet.positionY, rb+2);
+}
+
+void TankMove(PlayerPosition *player1, int choosePlayer){			/* pomicanje tenka prvog igraca */
+	int x,y;														/* prima  poziciju (srediste kruga) tenka i smjer prema kojem je okrenut */
+	x = player1->positionX;
+	y = player1->positionY;
+	
+	TankRemove(*player1);
+	if(choosePlayer == 0)
+		LCD_SetTextColor(LCD_COLOR_BLUE);
+	else
+		LCD_SetTextColor(LCD_COLOR_RED);
+	
+	if(player1->direction==1){							/* smjer 1=dolje, 2=lijevo, 3=gore, 4=desno */
+		if(player1->positionY+yt <= yScreenEnd && player1->positionY+yt >= yScreenStart)	
+			y=player1->positionY+yt;																																		
+		LCD_DrawFullCircle(x, y, r);
+		LCD_FillTriangle(x+a/2,x-a/2,x,y+r,y+r,y+r+a);
+	}
+	else if(player1->direction==2){
+		if(player1->positionX-xt <= xScreenEnd && player1->positionX-xt >= xScreenStart)
+			x=player1->positionX-xt;	
+		LCD_DrawFullCircle(x, y, r);
+		LCD_FillTriangle(x-r,x-r,x-r-a,y+a/2,y-a/2,y);
+	}
+	else if(player1->direction==3){
+		if(player1->positionY-yt <= yScreenEnd && player1->positionY-yt >= yScreenStart)
+			y=player1->positionY-yt;
+		LCD_DrawFullCircle(x, y, r);
+		LCD_FillTriangle(x-a/2,x+a/2,x,y-r,y-r,y-r-a);
+	}
+	else{
+		if(player1->positionX+xt <= xScreenEnd && player1->positionX+xt >= xScreenStart)
+			x=player1->positionX+xt;	
+		LCD_DrawFullCircle(x, y, r);
+		LCD_FillTriangle(x+r,x+r,x+r+a,y-a/2,y+a/2,y);
+	}
+	
+	player1->positionX = x;
+	player1->positionY = y;
+}
+
+void TankRemove(PlayerPosition player){			/* brisanje tenka */
+	LCD_SetTextColor(LCD_COLOR_WHITE);		/* prima poziciju tenka */
+	int x,y;
+	x=player.positionX;	
+	y=player.positionY;
+	if(player.direction==1){
+		LCD_DrawFullCircle(x, y, r);
+		LCD_FillTriangle(x+a/2,x-a/2,x,y+r,y+r,y+r+a);
+	}
+	else if(player.direction==2){
+		LCD_DrawFullCircle(x, y, r);
+		LCD_FillTriangle(x-r,x-r,x-r-a,y+a/2,y-a/2,y);
+	}
+	else if(player.direction==3){
+		LCD_DrawFullCircle(x, y, r);
+		LCD_FillTriangle(x-a/2,x+a/2,x,y-r,y-r,y-r-a);
+	}
+	else{
+		LCD_DrawFullCircle(x, y, r);
+		LCD_FillTriangle(x+r,x+r,x+r+a,y-a/2,y+a/2,y);
+	}
+}
+
```

Comment:
Možda glupo pitanje koje postavljam bez da previše razmišljam o njemu, ali zašto kad se briše tenk se ne može nacrtati primjerice bijeli kvadrat određene veličine?

Reply:
Nije losa ideja probat cemo to implementirat pa cemo vidit